src/platform/test.rs

```diff
@@ -770,3 +770,101 @@ mod sync_test {
         platform::OsIpcSender::test_not_sync();
     }
 }
+
+// Note! This test is actually used by the
+// cross_process_sender_transfer_spawn() test below.  Running it by
+// itself is meaningless.
+#[cfg(not(any(feature = "force-inprocess", target_os = "android")))]
+#[test]
+#[ignore]
+fn cross_process_two_step_transfer_server()
+{
+    let cookie: &[u8] = b"cookie";
+    let mut channel_name: Option<String> = None;
+
+    for arg in env::args() {
+        if arg.starts_with("channel_name:") {
+            let (_, name) = arg.split_at("channel_name:".len());
+            channel_name = Some(String::from(name));
+        }
+    }
+
+    if channel_name.is_none() {
+        return;
+    }
+
+    // connect by name to our other process
+    let super_tx = OsIpcSender::connect(channel_name.unwrap()).unwrap();
+
+    // create a channel for real communication between the two processes
+    let (sub_tx, sub_rx) = platform::channel().unwrap();
+
+    // send the other process the tx side, so it can send us the channels
+    super_tx.send(&[], vec![OsIpcChannel::Sender(sub_tx)], vec![]).unwrap();
+
+    // get two_rx from the other process
+    let (_, mut received_channels, _) = sub_rx.recv().unwrap();
+    assert_eq!(received_channels.len(), 1);
+    let two_rx = received_channels.pop().unwrap().to_receiver();
```

Comment:
Indeed, I forgot that `to_receiver()` requires `&mut` -- sorry for the confusion.